src/main/java/appeng/me/Grid.java

```diff
@@ -52,6 +52,7 @@ public Grid(final GridNode center) {
 
         final Map<Class<? extends IGridCache>, IGridCache> myCaches = AEApi.instance().registries().gridCache()
                 .createCacheInstance(this);
+        this.caches = new HashMap<>(myCaches.size());
```

Comment:
`IdentityHashMap` might be an idea here. Guava also provides a special class to instance map. Not sure if it might provide some additional benefits. E.g. creating a copy with new instances.